docs/conf.py

```diff
@@ -28,7 +28,7 @@
 project = 'icalendar'
 this_year = datetime.date.today().year
 copyright = f'{this_year}, Plone Foundation'
-version = pkg_resources.get_distribution('icalendar').version
+version = importlib.metadata.version('icalendar')
```

Comment:
You would usually open a new branch without the old commits by switching to master before. In your case though, I guess you would like to build with Python 3.12.